site/content/blog/2020-03-10-emojis.md

```diff
@@ -0,0 +1,77 @@
+---
+title: "All About Emojis"
+slug: all-about-emojis
+date: 2020-03-10T12:00:00-05:00
+categories:
+    - "markdown"
+author: Harrison Healey
+github: hmhealey
+community: harrison
+---
+
+Emojis are a great way to add tone to a piece of text. In person, you can easily tell someone's mood based on their body language and how they speak, but that's much more difficult with text alone. They also help make text-based conversation feel more casual, relaxed, and fun. Thanks to emojis, we can chat with much more real emotion than you might get by being careful about your word choice or by including just the right number of exclamation marks and periods at the end of a sentence.
+
+### So where did emojis come from?
+
+Emojis (or "emoji" which is technically the correct pluralization) are characters that are displayed as small images on devices that support them. This makes them different from emoticons which are loosely defined, purely text-based sequences of characters, such as ``<3`` or ``ಠ_ಠ``. The term emoji comes from the Japanese words e- (meaning "pictures") and -moji (meaning "character") since the original emojis were created by Japanese cell phone carriers in the 1990s. Eventually, the Unicode Consortium stepped in and came up with the standard set of emojis that we use today, although that set is frequently updated and expanded from the original version.
+
+{{< figure src="/blog/2020-03-10-emojis/emojis.png" alt="Examples of emojis" >}}
+
+### Emoji Sequences
+
+As Unicode characters, emojis are encoded on most platforms as UTF-8 characters which are rendered into images on platforms that support them. Emojis can also be made up of sequences containing more than one character, although they'll still be displayed as a single image as long as the emoji sequence is supported by the current platform. Some sequences are made by following an existing emoji with a special modifier character while others are made by combining multiple emojis.
+
+When typing and reading these characters, most devices will usually try to hide the fact that you're actually working on multiple characters, but when processing text with code, you have to be aware of these cases. Recently, I solved a bug involving this behaviour ([MM-22079](https://mattermost.atlassian.net/browse/MM-22079)). Currently, Mattermost tries to map Unicode emojis to the corresponding system emoji (which normally appears when using an emoji's short name such as `:taco:`), but when it was doing that, it would only look for the first character of the sequence, so it would accidentally display the wrong version of emojies. For example:
```

Comment:
```suggestion
When typing and reading these characters, most devices will usually try to hide the fact that you're actually working on multiple characters, but when processing text with code, you have to be aware of these cases. Recently, I solved a bug involving this behaviour ([MM-22079](https://mattermost.atlassian.net/browse/MM-22079)). Currently, Mattermost tries to map Unicode emojis to the corresponding system emoji (which normally appears when using an emoji's short name such as `:taco:`), but when it was doing that, it would only look for the first character of the sequence, so it would accidentally display the wrong version of emojis. For example:
```

Reply:
i just realized it's not obvious the change: emojies -> emojis